File: workspace/SRM145/div2_250/ImageDithering.cs
using System;
using System.IO;
using System.Collections.Generic;
using System.Text;

public class ImageDithering {
	public int count(string dithered, string[] screen) {
        int rtn = 0;

        for( int i = 0 ; i < screen.Length ; i++)
        {
            for( int j = 0; j < screen[i].Length; j++)
            {
                if (dithered.Contains(screen[i][j].ToString())){
                    rtn++;
                }
            }
        }
        return rtn;
	}

// CUT begin
	static bool DoTest(string dithered, string[] screen, int __expected) {
		DateTime startTime = DateTime.Now;
		Exception exception = null;
		ImageDithering instance = new ImageDithering();
		int __result = 0;
		try {
			__result = instance.count(dithered, screen);
		}
		catch (Exception e) { exception = e; }
		TimeSpan __elapsed = new TimeSpan(DateTime.Now.Ticks - startTime.Ticks);

		if (exception != null) {
			Console.Error.WriteLine("RUNTIME ERROR!");
			Console.Error.WriteLine(exception);
			return false;
		}
		else if (__result == __expected) {
			Console.Error.WriteLine("PASSED! " + string.Format("({0:0.00} seconds)", __elapsed.TotalSeconds));
			return true;
		}
		else {
			Console.Error.WriteLine("FAILED! " + string.Format("({0:0.00} seconds)", __elapsed.TotalSeconds));
			Console.Error.WriteLine("           Expected: " + __expected);
			Console.Error.WriteLine("           Received: " + __result);
			return false;
		}
	}

	public static void Main(string[] args){
		HashSet<int> cases = new HashSet<int>();
        for (int i = 0; i < args.Length; ++i) cases.Add(int.Parse(args[i]));

		Console.Error.WriteLine("ImageDithering (250 Points)");
		Console.Error.WriteLine();

    	int nCases = 0, nPassed = 0;
    	using (var reader = File.OpenText("ImageDithering.sample")) {
            while (true) {
                string line = reader.ReadLine();
                if (line == null || !line.StartsWith("--"))
                    break;

                string dithered = (string) Convert.ChangeType(reader.ReadLine(), typeof(string));
                string[] screen = new string[int.Parse(reader.ReadLine())];
                for (int i = 0; i < screen.Length; ++i)
                    screen[i] = (string) Convert.ChangeType(reader.ReadLine(), typeof(string));
                reader.ReadLine();
                int __answer = (int) Convert.ChangeType(reader.ReadLine(), typeof(int));

                nCases++;
                if (cases.Count > 0 && !cases.Contains(nCases - 1))
                	continue;
                Console.Error.Write(string.Format("  Testcase #{0} ... ", nCases - 1));
                if (DoTest(dithered, screen, __answer))
                    nPassed++;
            }
    	}

        if (cases.Count > 0) nCases = cases.Count;
		Console.Error.WriteLine();
		Console.Error.WriteLine(string.Format("Passed : {0}/{1} cases", nPassed, nCases));

		DateTime Jan1st1970 = new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
		int T = (int)((DateTime.UtcNow - Jan1st1970).TotalSeconds - 1513439302);
		double PT = T / 60.0, TT = 75.0;
		Console.Error.WriteLine(string.Format("Time   : {0} minutes {1} secs", T / 60, T % 60));
		Console.Error.WriteLine(string.Format("Score  : {0:0.00} points", 250 * (0.3 + (0.7 * TT * TT) / (10.0 * PT * PT + TT * TT))));
	}
// CUT end
}

File: workspace/SRM145/div2_250/ImageDithering.java
import java.io.*;
import java.util.*;

public class ImageDithering {
	public int count(String dithered, String[] screen) {
		int rtn = 0;

		for( int j = 0 ; j < screen.length ; j++ ){
			char[] charArray = screen[j].toCharArray();
			for( char ch : charArray ){
				if( dithered.contains(String.valueOf(ch))){
					rtn++;
				}
			}
		}
		return rtn;
	}

// CUT begin
	public static void main(String[] args){
		System.err.println("ImageDithering (250 Points)");
		System.err.println();
		HashSet<Integer> cases = new HashSet<Integer>();
        for (int i = 0; i < args.length; ++i) cases.add(Integer.parseInt(args[i]));
        runTest(cases);
	}

	static void runTest(HashSet<Integer> caseSet) {
	    int cases = 0, passed = 0;
	    while (true) {
	    	String label = Reader.nextLine();
	    	if (label == null || !label.startsWith("--"))
	    		break;

            String dithered = Reader.nextLine();
            String[] screen = new String[Integer.parseInt(Reader.nextLine())];
            for (int i = 0; i < screen.length; ++i)
                screen[i] = Reader.nextLine();
            Reader.nextLine();
            int __answer = Integer.parseInt(Reader.nextLine());

            cases++;
            if (caseSet.size() > 0 && !caseSet.contains(cases - 1))
                continue;
    		System.err.print(String.format("  Testcase #%d ... ", cases - 1));

            if (doTest(dithered, screen, __answer))
                passed++;
	    }
	    if (caseSet.size() > 0) cases = caseSet.size();
        System.err.println(String.format("%nPassed : %d/%d cases", passed, cases));

        int T = (int)(System.currentTimeMillis() / 1000) - 1513439308;
        double PT = T / 60.0, TT = 75.0;
        System.err.println(String.format("Time   : %d minutes %d secs%nScore  : %.2f points", T / 60, T % 60, 250 * (0.3 + (0.7 * TT * TT) / (10.0 * PT * PT + TT * TT))));
	}

	static boolean doTest(String dithered, String[] screen, int __expected) {
        dithered = new String(dithered);
		for (int i = 0; i < screen.length; i++) {
			screen[i] = new String(screen[i]);
		}
		long startTime = System.currentTimeMillis();
		Throwable exception = null;
		ImageDithering instance = new ImageDithering();
		int __result = 0;
		try {
			__result = instance.count(dithered, screen);
		}
		catch (Throwable e) { exception = e; }
		double elapsed = (System.currentTimeMillis() - startTime) / 1000.0;

		if (exception != null) {
			System.err.println("RUNTIME ERROR!");
			exception.printStackTrace();
			return false;
		}
		else if (__result == __expected) {
			System.err.println("PASSED! " + String.format("(%.2f seconds)", elapsed));
			return true;
		}
		else {
			System.err.println("FAILED! " + String.format("(%.2f seconds)", elapsed));
			System.err.println("           Expected: " + __expected);
			System.err.println("           Received: " + __result);
			return false;
		}
	}

	static class Reader {
        private static final String dataFileName = "ImageDithering.sample";
	    private static BufferedReader reader;

	    public static String nextLine() {
	        try {
                if (reader == null) {
                    reader = new BufferedReader(new InputStreamReader(new FileInputStream(dataFileName)));
                }
                return reader.readLine();
	        }
	        catch (IOException e) {
	            System.err.println("FATAL!! IOException");
	            e.printStackTrace();
	            System.exit(1);
	        }
	        return "";
	    }
	}
// CUT end
}
